### DISCUSSION TITLE
Missing LINQ QueryDSL with Scala

### DISCUSSION THREAD

**Discussion by ghost** [scala/scala3#17902] [Feature Requests]
Hello, I know scala have Slick , but it's too ugly than C# LINQ.
Here is the Scala SLICK syntax:
```
val q = for {
    c <- Coffees if c.supID === 101
} yield (c.name, c.price)
```
This would be equivalent to the following C#:

```
// LINQ extension method syntax
var q = Coffees.Where(c => c.supID == 101)
               .Select(c => new { c.name, c.Price});

// LINQ query syntax
var q = from c in Coffees 
          where c.supId == 101 
          select new { c.name, c.price};
```
Yield and Select appear to be the same sort of projection routine.

Personally, I think the Scala one feels ugly, but that might only be because both C# options are so readable.

Now Groovy have GINQ , Looks good.
```
import groovy.transform.*
@TupleConstructor
@EqualsAndHashCode
@ToString
class Person {
    String name
    String nickname
}

def linda = new Person('Linda', null)
def david = new Person('David', null)
def persons = [new Person('Daniel', 'ShanFengXiaoZi'), linda, david]
def result = GQ {
    from p in persons
    where p.nickname == null
    select p
}.stream()
    .peek(p -> { p.nickname = 'Unknown' }) // update `nickname`
    .toList()

def expected = [new Person('Linda', 'Unknown'), new Person('David', 'Unknown')]
assert expected == result
assert ['Unknown', 'Unknown'] == [linda, david]*.nickname // ensure the original objects are updated

```
So i wonder if Scala 3 can support some feature like LINQ, Or learn from Groovy lang.

The QueryDSL was the big missing part of Scala .

http://mail-archives.apache.org/mod_mbox/groovy-notifications/201907.mbox/%3CJIRA.13237843.1559779174000.16465.1563559080283@Atlassian.JIRA%3E

[jira] [Updated] (GROOVY-9159) [GEP] Support LINQ, aka GINQ

https://docs.groovy-lang.org/docs/next/html/documentation/#_querying_collections_in_a_sql_like_style

**Comment by guizmaii** [comment]
@netroby Do you know https://github.com/getquill/quill? It might be what you're looking for. It's closer to LINQ than Slick, I think.
